language/bazel/visibility/lang.go

```diff
@@ -65,9 +67,22 @@ func (*visibilityExtension) GenerateRules(args language.GenerateArgs) language.G
 	}
 
 	r := rule.NewRule("package", "")
-	r.SetAttr("default_visibility", cfg.visibilityTargets)
+	for _, er := range args.File.Rules {
+		if er.Kind() == "package" {
+			r = er
+			break
+		}
+	}
+
+	if len(cfg.visibilityTargets) > 0 {
+		r.SetAttr("default_visibility", cfg.visibilityTargets)
+	}
+	if len(cfg.features) > 0 {
+		r.SetAttr("features", cfg.features)
+	}
 
-	insertIndex := 0
+	// Start after the first statements if no rules exist
+	insertIndex := len(args.File.File.Stmt)
```

Comment:
partially unrelated change here as an oversight in https://github.com/bazel-contrib/bazel-gazelle/pull/2249 that i discovered in these new tests, if the file has no rules `0` isn't really nice because it put them above the gazelle comments.

i can split this into a later pr if you want

Reply:
This is fine as it is.